invenio_oauthclient/contrib/cern.py

```diff
@@ -209,12 +209,19 @@ def get_resource(remote):
 def account_info(remote, resp):
     """Retrieve remote account information used to find local user."""
     res = get_resource(remote)
+    session.pop('cern_resource')
 
     email = res['EmailAddress'][0]
     external_id = res['PersonID'][0]
     nice = res['CommonName'][0]
     name = res['DisplayName'][0]
 
+    groups = fetch_groups(res['Group'])
+    provides = [UserNeed(email)] + \
+               [RoleNeed('{0}@cern.ch'.format(group)) for group in groups]
+
+    session['identity.cern_provides'] = provides
```

Comment:
Not sure if this is safe to put it to the session - the user has not been authenticated at this point in time.